terraform/modules/data-share-service/github_iam.tf

```diff
@@ -0,0 +1,81 @@
+data "aws_iam_policy_document" "github_oidc_environment_assume" {
+  statement {
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    principals {
+      type = "Federated"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"
+      ]
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = ["sts.amazonaws.com"]
+      variable = "token.actions.githubusercontent.com:aud"
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = ["repo:alphagov/gdx-data-share-poc:environment:${var.environment}"]
+      variable = "token.actions.githubusercontent.com:sub"
+    }
+  }
+}
+
+resource "aws_iam_role" "github_oidc_environment" {
+  name               = "${var.environment}-github-oidc-deploy"
+  assume_role_policy = data.aws_iam_policy_document.github_oidc_environment_assume.json
+}
+
+data "aws_iam_policy" "github_oidc_environment" {
+  name = "Admin"
```

Comment:
I know this isn't a change, but I find this very scary

Reply:
I am happy for us to lock this down if you know what terraform needs